src/libraries/System.Text.Json/tests/System.Text.Json.Tests/JsonNode/JsonObjectTests.cs

```diff
@@ -1060,6 +1060,59 @@ public static void DeepEquals()
             Assert.False(JsonNode.DeepEquals(jObject, diffJObject));
         }
 
+        [Fact]
+        public static void DeepEquals_JsonObject_With_JsonValuePOCO()
+        {
+            var jObject = new JsonObject();
+            jObject["Id"] = 1;
+            jObject["Name"] = "First";
+            var nestedObject = new JsonObject();
+            nestedObject["Id"] = 2;
+            nestedObject["Name"] = "Last";
+            nestedObject["NestedObject"] = null;
+            jObject["NestedObject"] = nestedObject;
+
+            var poco = new SimpleClass()
+            {
+                Id = 1,
+                Name = "First",
+                NestedObject = new SimpleClass()
+                {
+                    Id = 2,
+                    Name = "Last",
+                }
+            };
+
+            Assert.True(JsonNode.DeepEquals(jObject, JsonValue.Create(poco)));
+        }
+
+        [Fact]
+        public static void DeepEquals_JsonObject_With_Dictionary()
+        {
+            var jObject = new JsonObject();
+            jObject["One"] = 1;
+            jObject["array"] = new JsonArray() { "a", "b" };
+            jObject["obj"] = new JsonObject();
+
+            var dictionary = new Dictionary<string, object>()
+            {
+                { "One", 1 },
+                { "array", new string[] { "a", "b" } },
+                { "obj", new { } }
+            };
+
+            Assert.True(JsonNode.DeepEquals(jObject, JsonValue.Create(dictionary)));
```

Comment:
Likewise here, I'd like a test case checking a dictionary that isn't equal (e.g. because it has one additional entry).